### ISSUE TITLE
BLID & Password

### ISSUE THREAD

**Issue by derfletcher** [home-assistant/home-assistant.io#30869]
### Feedback

Hi, I am sorry, but this documentation is to rough for a non-developer. 
I had several month my irobot integration up and running, without issues. Then from one day to another, the integration is asking me for a "BLID" and a Password, which I somehow need to get from the irobot cloud by installing dorita980 and running it...

I have no clue how this works. Isn´t it possible to integrate the service into the frontend configuration of home assistant? 
Home assistant is best application for non-technical people who want to make their home smart, but here it has a huge gap.
Please help and simpliy the frontend solution and fix the issue of connecting irobot to home assistant!

Thanks
Thiemo

### URL

https://www.home-assistant.io/integrations/roomba/

### Version

2024.1.3

### Additional information

I have a roomba j9+ and a Braava M6

**Comment by derfletcher** [comment]
This issue is still valid and needed! 
Thanks
Thiemo

**Comment by cryocide** [comment]
Having the same issue here. The integration attempts to pull the info during a robot reboot and fails and recommends that I do this manually, but the steps listed after the link make no sense in the context that I'm running a pre-built home-assistant image on RaspberryPi. I could probably fumble through getting the tools on another system, or figure out how to manually tunnel into the image and do it myself there, but that feels like it's outside the scope of what home-assistant is supposed to be.

**Comment by pgroene** [comment]
I try to run the password retrival in the homeassisant docker container and I get the following error:

```
docker exec -it home-assistant python -c 'import roombapy.entry_points; roombapy.entry_points.password()' 192.168.0.60
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'roombapy.entry_points'
```

What is the command I need to run from the command line? It seems the import is not correct or not longer included with homeassistant

**Comment by pgroene** [comment]
I think the correct command is:
```
docker exec -it home-assistant bash -c 'roombapy discover 192.168.0.60'

```

However it seems gives an error:
Note: Password for Roomba couldn't be obtained.

**Comment by wangjingfei** [comment]
Same issue for me. Anyone got the solution?

**Comment by derfletcher** [comment]
This issue is still valid and a solution is still needed!
Thanks
Thiemo

**Comment by Mizukamakiri** [comment]
Definitely still an issue. `docker exec -it homeassistant bash -c 'roombapy discover 'IP_ADDRESS'` works, and returns everything except the password.

**Comment by pgroene** [comment]
I can post my command tomorrow when I have access to my machine to get the passwords. I used an other docker container for retrieving them

**Comment by pgroene** [comment]
$ git clone https://github.com/koalazak/dorita980.git
$ cd dorita980
$ npm install
$ npm run get-password-cloud <iRobot Username> <iRobot Password> [Optional API-Key]

**Comment by Mizukamakiri** [comment]
Thought i'd update with what worked for me, in case others come looking:
` docker run -it node sh -c "npm install -g dorita980 && get-roomba-password-cloud YOUR_IROBOT_USERNAME YOUR_IROBOT_PASSWORD`

**Comment by holocronweaver** [comment]
To retrieve the password for a Roomba i7 I had to use the dorita980 app, as pgroene and Mizukamakiri suggest. Afterwards I had to reboot the Roomba (push and hold the 'Clean' button for 15 seconds) to get the iRobot app to successfully issue cleaning jobs again.

**Comment by cryocide** [comment]
Since my last comment, I've migrated to a HAOS build in a VM. Still can't find a solution to this in the documentation that works with a HAOS build. Also, the configuration wizard offers instructions for a different model of Roomba that doesn't work on a J7, and the instructions that are given don't explain what it is that I'm trying to make the robot do so I can't look up the correct button/dock sequence to make it restart or attempt pairing or whatever. Some model-specific instructions might be helpful, along with a workaround that works on HomeAssistantOS.

**Comment by Mizukamakiri** [comment]
> Since my last comment, I've migrated to a HAOS build in a VM. Still can't find a solution to this in the documentation that works with a HAOS build. Also, the configuration wizard offers instructions for a different model of Roomba that doesn't work on a J7, and the instructions that are given don't explain what it is that I'm trying to make the robot do so I can't look up the correct button/dock sequence to make it restart or attempt pairing or whatever. Some model-specific instructions might be helpful, along with a workaround that works on HomeAssistantOS.

If you use my solution, there are no buttons to press. (the username and password, is what you use to login to irobot.com)

**Comment by derfletcher** [comment]
I think that all the comments here Show, that this process is even for technical experienced people difficult. A Build in Solution Soul be great!

**Comment by ffbe42** [comment]
> Definitely still an issue. `docker exec -it homeassistant bash -c 'roombapy discover 'IP_ADDRESS'` works, and returns everything except the password.

I have a Roomba 960,  this works for me  and it DOES return the password. Before you run the command you need to press the home button on top of the Roomba for 2 seconds (so that the Roomba beeps). Then run the docker  -it command.

**Comment by cryocide** [comment]
> I think that all the comments here Show, that this process is even for technical experienced people difficult. A Build in Solution Soul be great!

Agreed. I think the best way for devs to think of it in this sense would be, “How would I do this on Home Assistant Green?” Docker commands aren’t really accessible in that scenario.

**Comment by willmackey2nd** [comment]
I'm a dev but still every time I need to visit this page I feel it's messy, complicated and targeted to devs that know how to fill in the missing info.
It also makes weird assumptions that most would be running HA in a Docker container. I'm running HAOS VM in Proxmox and none of these Python scripts exist there. Then needing to install npm or docker to run this script that could be built in the integration feels just weird.

At least should add clear steps for people not working on a linux machine with Docker and/or npm installed.

**Comment by elafontaine** [comment]
Same issue, still valid today

**Comment by Airmaxx014** [comment]
https://community.home-assistant.io/t/roomba-j7-integration-help/339705/72

Look at this item
That worked for me … I used the app Hodor and did take the trail for 3 days. works a little different then Thor but the output is the same.

1 - install the app
2 - Open the app and install the certificate
4 - Allow trust
5 - start the app
6 - open Irobot app
7 - stop the Hodor app with the sniffer.
8 - Look for Unauth x    v2/login
9 - open the application/json file

And there you find the password.

I did use a Iphone. But will work on Android . But then you need to find another app that can sniffer

**Comment by holocronweaver** [comment]
Still an issue.

**Comment by Airmaxx014** [comment]
Just try to use my idea. That is working. Then there is no issue.

**Comment by holocronweaver** [comment]
@Airmaxx014 Feel free to open a PR to update the docs and code if you think you have found a solution for the masses. This issue is to create a proper user experience without workarounds, then document them in the official HA docs. That way there is a documented, quick, dependable and relatively painless / tech-lite way to do this, as with most integrations.

**Comment by holocronweaver** [comment]
Still at issue. 

Hey there @pschmitt, @cyr-ius, @shenxn, @Xitee1, have any of you haha a chance to take a look?

**Comment by maxaalderink** [comment]
The documentation of the paragraph [Retrieving your credentials](https://www.home-assistant.io/integrations/roomba/#retrieving-your-credentials) is still outdated and should be changed to what @Mizukamakiri commented [here](https://github.com/home-assistant/home-assistant.io/issues/30869#issuecomment-2359157545).

Although I only had success with the instruction of paragraph [Retrieving credentials from the cloud with dorita980](https://www.home-assistant.io/integrations/roomba/#retrieving-credentials-from-the-cloud-with-dorita980)

**Comment by dran0** [comment]
> > Definitely still an issue. `docker exec -it homeassistant bash -c 'roombapy discover 'IP_ADDRESS'` works, and returns everything except the password.
> 
> I have a Roomba 960, this works for me and it DOES return the password. Before you run the command you need to press the home button on top of the Roomba for 2 seconds (so that the Roomba beeps). Then run the docker -it command.

Thank you so much this worked for my S9+.

**Comment by holocronweaver** [comment]
Bump

**Comment by holocronweaver** [comment]
Still an issue.

**Comment by holocronweaver** [comment]
Still an issue.

**Comment by holocronweaver** [comment]
Still an issue.
